src/encoded/types/shared_calculated_properties.py

```diff
@@ -482,85 +482,85 @@ class CalculatedReplicationType:
+        for r in filtered_replicate_objects:
+            bio_rep_num = r['biological_replicate_number']
             if bio_rep_num not in bio_rep_dict:
-                bio_rep_dict[bio_rep_num] = replicate_object
+                bio_rep_dict[bio_rep_num] = r
                 continue
-            tech_rep_num = replicate_object['technical_replicate_number']
+            tech_rep_num = r['technical_replicate_number']
             if tech_rep_num < bio_rep_dict[bio_rep_num]['technical_replicate_number']:
-                bio_rep_dict[bio_rep_num] = replicate_object
-
-        # Compare the biosamples to see if for humans they are the same donor and for
-        # model organisms if they are sex-matched and age-matched
-        biosample_donor_list = []
-        biosample_number_list = []
-
-        for replicate_object in bio_rep_dict.values():
-            if 'libraries' in replicate_object and replicate_object['libraries']:
-                biosamples = request.select_distinct_values(
-                    'biosample', *replicate_object['libraries']
-                )
-                if biosamples:
-                    for b in biosamples:
-                        biosample_object = request.embed(b, '@@object')
-                        biosample_donor_list.append(
-                            biosample_object.get('donor')
-                        )
-                        biosample_number_list.append(
-                            replicate_object.get('biological_replicate_number')
-                        )
-                        biosample_species = biosample_object.get('organism')
-                        biosample_type_object = request.embed(
-                            biosample_object['biosample_ontology'],
-                            '@@object'
-                        )
-                        biosample_type = biosample_type_object.get('classification')
-                else:
-                    # special treatment for "RNA Bind-n-Seq" they will be called unreplicated
-                    # untill we change our mind
-                    if assay_term_name == 'RNA Bind-n-Seq':
-                        return 'unreplicated'
-                    # If I have a library without a biosample,
-                    # I cannot make a call about replicate structure
-                    return None
-            else:
-                # REPLICATES WITH NO LIBRARIES WILL BE CAUGHT BY AUDIT (TICKET 3268)
-                # If I have a replicate without a library,
-                # I cannot make a call about the replicate structure
+                bio_rep_dict[bio_rep_num] = r
+        biosample_donors_set = set()
+        biosample_numbers_set = set()
+        for r in bio_rep_dict.values():
+            library = r.get('library')
+            if not library:
+                # If I have a replicate without a library I cannot make a
+                # call about the replicate structure.
                 return None
```

Comment:
Can use `any()` to simplify this a little:
```suggestion
        if any(not r.get('library') for r in bio_rep_dict.values():
            # If I have a replicate without a library I cannot make a
            # call about the replicate structure.
            return None
```

Reply:
are you saying it is better to split the thing into two for loops? once for libraries in replicates and aother for biosamples in these libraries?

Reply:
No I am just saying we can simplify the for loop over the `bio_rep_dict` using `any`

Reply:
discussed on slack and convinced Paul it is OK :)